addons/mail/static/src/models/thread/thread.js

```diff
@@ -149,6 +149,10 @@ function factory(dependencies) {
          */
         static convertData(data) {
             const data2 = {};
+            if ('channel_id' in data) {
+                data2.id = data.channel_id;
+                data2.model = 'mail.channel';
+            }
```

Comment:
This is related to previous comment. This should be changed so that the bus notification contains proper thread model and id, instead of channel_id.

All other handlers/formatters make the abstraction of `channel.partner` in regard to the JS, this one should not be an exception.

Reply:
After having issue on my own PR with this. I think it's not possible as it is due to the need of knowing the `id` of `channel.partner` to properly insert the `partnerSeenInfos`.

However, `ThreadPartnerSeenInfo` could be changed so that the `find` method is using `thread` and `partner` as "primary key" instead of the `channel.partner` `id`.

Reply:
If need be due to issue with passing commands, we'll need to create thread_id and partner_id fields from which the relations will be computed.

Reply:
> However, ThreadPartnerSeenInfo could be changed so that the find method is using thread and partner as "primary key" instead of the channel.partner id.

That was what was done before but we went with id instead as in find function you don't get plain fields for relation but commands. So why couldn't we use the id but make sur that ThreadPartnerSeenInfo model handles that instead of Thread ? or maybe use the same function as if it was a "mail.channel" notif ?

Reply:
I'm not sure to understand your message, but I've figured that the ID can be crafted manually in JS by assembling channel id and partner id together. Then no need for the python to "leak" the channel.partner ID anymore.

Reply:
What I was saying (and I admit I wasn't clear) is that in function `_createRecordLocalId` when you want to use channel and partnerd id you don't get these id, you get relation commands (like `[["insert", { ... }]]` or `[["link, ... ]]`) so it's difficult to be able to use these fields to build the id

Reply:
I think you meant `_findFunctionFromData` not `_createRecordLocalId`.
But I understand the issue. You want partner and thread "id" instead of "link commands". This is actually possible.

Either:
- manually craft the "id" of the MessageSeenIndicator (instead of using the one from channel partner) based on `${channel_id}_${partner_id}`
- add 2 new fields, `partner_id` and `channel_id`. Then you write those as id instead of writing the relation. And the relation can be a compute based on this field (onchange doing the link).